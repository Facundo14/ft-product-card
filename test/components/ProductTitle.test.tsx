import React from 'react';
import renderer from 'react-test-renderer';
import { ProductTitle, ProductCard } from '../../src/components';
import { product1 } from '../data/products';


describe('ProductImage', () => {

    test('debe de mostrar el componente correctamente con la imagen personalizada', () => {
        const wrapper = renderer.create(
            <ProductTitle title="Custom product" className='custom-class' />
        );

        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    test('debe mostrar el componente con la imagen del producto', () => { 
        const wrapper = renderer.create(
            <ProductCard product={ product1 } >
                {
                    () => (
                        <ProductTitle />
                    )
                }
            </ProductCard>
        );

        expect(wrapper.toJSON()).toMatchSnapshot();
     })
});