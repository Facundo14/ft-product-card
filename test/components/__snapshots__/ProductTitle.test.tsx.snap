// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage debe de mostrar el componente correctamente con la imagen personalizada 1`] = `
<span
  className="productDescription custom-class"
>
  Custom product
</span>
`;

exports[`ProductImage debe mostrar el componente con la imagen del producto 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffee Mug - Sin Imagen
  </span>
</div>
`;
